source/dem/write_checkpoint.cc

```diff
@@ -67,6 +73,21 @@ write_checkpoint(
     {
       solid_objects[i]->write_checkpoint(prefix);
     }
+
+  // Prepare the checkpoint controller for checkpointing
+  // We don't use the same prefix, since this file needs to have the same name
+  // regardless of the checkpoint id being use. This file is giving the
+  // information of which checkpoint id to use when restarting
+  std::string checkpoint_controller_object_filename =
+    parameters.restart.filename + ".checkpoint_controller";
+  std::ofstream oss_checkpoint_controller_obj(
+    checkpoint_controller_object_filename);
+  boost::archive::text_oarchive oa_checkpoint_controller_obj(
+    oss_checkpoint_controller_obj, boost::archive::no_header);
+  checkpoint_controller.serialize(oa_checkpoint_controller_obj, 0);
+
+  // Increment for the next checkpoint
+  checkpoint_controller.increment_checkpoint_id();
```

Comment:
Why not increment the id after you have written?

Reply:
That is what is being done at line 90. The last line of the function, thus after the writing.

Reply:
Yes but why not do it wihtin the class?
Let's talk about it tomorrow